tests/openedx_tagging/core/tagging/test_views.py

```diff
@@ -376,3 +379,242 @@ def test_delete_taxonomy_404(self):
+        # to 200 ObjectTags
+        self.open_taxonomy_enabled = Taxonomy.objects.create(
+            name="Enabled Free-Text Taxonomy"
+        )
+        self.open_taxonomy_disabled = Taxonomy.objects.create(
+            name="Disabled Free-Text Taxonomy", enabled=False
+        )
+        for i in range(200):
+            ObjectTag.objects.create(
+                object_id="abc", taxonomy=self.open_taxonomy_enabled, _value=f"Free Text {i}"
+            )
+            ObjectTag.objects.create(
+                object_id="abc", taxonomy=self.open_taxonomy_disabled, _value=f"Free Text {i}"
+            )
+
+    @ddt.data(
+        (None, "abc", status.HTTP_403_FORBIDDEN, None, None),
+        ("user", "abc", status.HTTP_200_OK, 421, 10),
+        ("staff", "abc", status.HTTP_200_OK, 421, 10),
+        (None, "non-existing-id", status.HTTP_403_FORBIDDEN, None, None),
+        ("user", "non-existing-id", status.HTTP_200_OK, 0, 0),
+        ("staff", "non-existing-id", status.HTTP_200_OK, 0, 0),
+    )
+    @ddt.unpack
+    def test_retrieve_object_tags(
+        self, user_attr, object_id, expected_status, expected_count, expected_results
+
+    ):
+        """
+        Test retrieving object tags
+        """
+        url = OBJECT_TAGS_RETRIEVE_URL.format(object_id=object_id)
+
+        if user_attr:
+            user = getattr(self, user_attr)
+            self.client.force_authenticate(user=user)
+
+        response = self.client.get(url)
+        assert response.status_code == expected_status
+
+        if status.is_success(expected_status):
+            assert response.data.get("count") == expected_count
+            assert response.data.get("results") is not None
+            assert len(response.data.get("results")) == expected_results
+
+    @ddt.data(
+        (None, "abc", status.HTTP_403_FORBIDDEN, None, None),
+        ("user", "abc", status.HTTP_200_OK, 20, 10),
+        ("staff", "abc", status.HTTP_200_OK, 20, 10),
+    )
+    @ddt.unpack
+    def test_retrieve_object_tags_taxonomy_queryparam(
+        self, user_attr, object_id, expected_status, expected_count, expected_results
+    ):
+        """
+        Test retrieving object tags for specific taxonomies provided
+        """
+        url = OBJECT_TAGS_RETRIEVE_URL.format(object_id=object_id)
+
+        if user_attr:
+            user = getattr(self, user_attr)
+            self.client.force_authenticate(user=user)
+
+        response = self.client.get(url, {"taxonomy": self.enabled_taxonomy.pk})
+        assert response.status_code == expected_status
+        if status.is_success(expected_status):
+            assert response.data.get("count") == expected_count
+            assert response.data.get("results") is not None
+            assert len(response.data.get("results")) == expected_results
+            object_tags = response.data.get("results")
+            for object_tag in object_tags:
+                assert object_tag.get("is_valid") is True
```

Comment:
I initially had a test case to for `is_valid=False`, however after some recently added uniqueness constraints ([diff](https://github.com/openedx/openedx-learning/commit/fb1fd0860f4873aae9250901a6af75a5f86384cc#diff-7ffb9100f9426c81c59e88e9beceb9e3a8089da152d1d605b7de97990e6f0d50R546)) on the db level that made it harder to "create" invalid object tags, but I figured it out and added an invalid object tag check.

Reply:
Ah, I meant an invalid tag which has been deleted from the closed taxonomy.

1. Create a taxonomy or just use the existing `"Enabled Taxonomy"`
2. Create a new tag in the taxonomy called "deleted tag"
3. Tag some object(s) with that tag
4. Delete the tag from the taxonomy.
5. Those objects should now have the "deleted tag" but it should be invalid, because it no longer exists in the taxonomy.

I think ^ is a very realistic scenario that definitely will happen, and that's why I want to test for it.
I think creating a tag with no value is not a realistic scenario, if I understand your current version of the new test correctly.

CC @pomegranited, make sure the scenario I outlined above is an accurate description of a way to get invalid tags.

Reply:
@bradenmacdonald @pomegranited I see, thanks for the clarification! I added updated the test to account for the scenario described above.